packages/react/accordion/src/Accordion.tsx

```diff
@@ -222,21 +230,22 @@ interface AccordionImplProps extends PrimitiveDivProps {
 const AccordionImpl = React.forwardRef<AccordionImplElement, AccordionImplProps>(
   (props: ScopedProps<AccordionImplProps>, forwardedRef) => {
     const { __scopeAccordion, disabled, ...accordionProps } = props;
-    const triggerNodesRef = React.useRef<Set<AccordionTriggerElement>>(new Set());
     const accordionRef = React.useRef<AccordionImplElement>(null);
     const composedRefs = useComposedRefs(accordionRef, forwardedRef);
+    const getItems = useCollection(__scopeAccordion);
 
     const handleKeyDown = composeEventHandlers(props.onKeyDown, (event) => {
       const target = event.target as HTMLElement;
       const isAccordionKey = ACCORDION_KEYS.includes(event.key);
+      const triggerCollection = getItems();
 
       if (!isAccordionKey || !isButton(target)) {
```

Comment:
I realised we have a pretty nasty assumption here (that the trigger is always a `HTMLButtonElement`). It's probably fine in most cases but if someone used `asChild` to change the node, our keyboard accessibility logic would break. Shall I create a ticket to look into this?

Reply:
What was the `isButton` check for in the first place?
Was it to make sure the event was in fact bubbling from an accordion trigger?
If so, regarding of what tag it is, isn't it better to just check that it's part of the triggers?
It seems it was in fact already handled before by the `if (triggerIndex === -1) return;` check?

I might be completely out of it though 😀

Reply:
Oh yeah... thank god you're here 😅 

I'll remove that `isButton` check as part of this PR while I'm here then.